src/editor-registry.coffee

```diff
@@ -0,0 +1,35 @@
+{Emitter, Disposable} = require 'event-kit'
+
+# This global registry tracks registered `TextEditors`.
+#
+# Packages that provide extra functionality to `TextEditors`, such as
+# autocompletion, can observe this registry to find applicable editors.
+module.exports =
+class EditorRegistry
```

Comment:
Small nitpick, but I'd really like this to be called `TextEditorRegistry` for consistency with the `TextEditor` class. I know I proposed calling the global `atom.editors`, but I wonder if we *should* actually call it `atom.textEditors` for more precision and consistency.

I originally wanted to keep it one word so it looked better, but now that I think about it, it would be the first unqualified use of the word "editor" in our API. Many different kinds of editors could potentially exist within Atom, so maybe we should be clear that we're talking about text editors. Would you be down to change this? Thanks.